doc/read-the-docs-site/doc/marconi-as-a-library/tutorials/BasicApp.hs

```diff
@@ -0,0 +1,484 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Main where
+
+-- BLOCKSTART import
+
+import Control.Concurrent (MVar)
+import Control.Concurrent.Async (race_)
+import Control.Lens ((^.))
+import Control.Monad.Except (ExceptT, MonadError)
+import Control.Monad.IO.Class (MonadIO, liftIO)
+import Data.Aeson (FromJSON, ToJSON, (.=))
+import Data.Aeson qualified as Aeson
+import Data.List qualified as List
+import Data.Maybe (listToMaybe)
+import Data.Text (Text)
+import Database.SQLite.Simple qualified as SQL
+import Database.SQLite.Simple.QQ (sql)
+import Database.SQLite.Simple.ToField qualified as SQL
+import GHC.Generics (Generic)
+import Network.Wai.Handler.Warp (defaultSettings, runSettings, setPort)
+import Servant (Application, Proxy (Proxy), Server, serve, (:>))
+import System.Environment (getArgs)
+
+import Cardano.Api qualified as C
+import Cardano.BM.Setup qualified as Trace
+import Cardano.BM.Tracing qualified as Trace
+import Marconi.ChainIndex.Experimental.Indexers.Orphans ()
+import Marconi.ChainIndex.Experimental.Indexers.SyncHelper qualified as Core
+import Marconi.ChainIndex.Experimental.Runner qualified as Core
+import Marconi.ChainIndex.Node.Client.Retry qualified as Core
+import Marconi.ChainIndex.Orphans ()
+import Marconi.ChainIndex.Types (BlockEvent (BlockEvent), SecurityParam)
+import Marconi.ChainIndex.Utils qualified as Utils
+import Marconi.Core qualified as Core
+import Marconi.Core.JsonRpc qualified as Core
+import Network.JsonRpc.Types (JsonRpc, RawJsonRpc)
+
+-- BLOCKEND import
+
+main :: IO ()
+main = runBlockInfoSqliteIndexerHttp
+
+-- BLOCKSTART env
+data Env indexer = Env
+  { envTrace :: !(Trace.Trace IO Text)
+  -- ^ The 'Trace' used for logging
+  , envSocketFilePath :: !String
+  -- ^ The local file path of the local node's socket file
+  , envNetworkId :: !C.NetworkId
+  -- ^ Network Id of the local node
+  , envSecurityParam :: !SecurityParam
+  -- ^ Security parameter (K) of the local node
+  , envIndexerWorker :: !(Core.WorkerIndexer IO BlockEvent BlockInfoEvent indexer)
+  -- ^ Worker of the indexer we want to run.
+  --
+  -- * 'BlockEvent': the data we get from the local node.
+  -- * 'BlockInfoEvent': the event type
+  -- * 'indexer' the indexer backend (ListIndexer, SQLiteIndexer, FileIndexer, etc.)
+  }
+
+withIndexerBuildEnv
+  :: Core.WorkerIndexer IO BlockEvent BlockInfoEvent indexer
+  -> (Env indexer -> IO ())
+  -> IO ()
+withIndexerBuildEnv worker action = do
+  -- We start by getting the network id and socket file path of the local node.
+  -- We get the values through the CLI.
+  [socketFilePath, networkMagicStr] <- getArgs
+  let networkMagic = read networkMagicStr
+
+  -- The security parameter.
+  -- TODO Should be queried from the local node.
+  let k = 2_160
```

Comment:
TODO's like this can be a bit confusing inside the tutorial, could you maybe elaborate a bit? For example, you could say that, for simplicity, we hardcode the security parameter (I'm not sure if this is the reason but that's why I think this needs some more detailing).

Reply:
I changed it by actually implementing the TODO: querying the local node for the security param. Let me know what you think.